discovery/gossiper.go

```diff
@@ -1624,6 +1623,7 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 				d.bestHeight)
 			d.Unlock()
 			return nil
+			nMsg.err <- nil
```

Comment:
Interesting, I think the _original_ idea here was to block the caller until the next block arrived which then would allow us to process the announcement. IMO we really need to just get rid of this things. It _is_ an unbounded map after all...

Reply:
Also why not return the error, then return nil?

Reply:
One thing that thing that comes to mind re the areas that we send back an error over a channel: we should look into making a higher order function that takes a function and an error channel, then calls the function and returns the error over the channel. This way, we don't always need to remember to _both_ return the error and send over the channel, thereby reducing the cognitive burden on any one modifying these areas of the codebase.

Reply:
> Interesting, I think the original idea here was to block the caller until the next block arrived which then would allow us to process the announcement.

Correct. Having the channel send should've been included in my changes when I removed the premature announcement map. I went through the rest of this method and made sure there weren't any other cases.

> Also why not return the error, then return nil?

Somehow this got messed up while I was staging the changes for the commit. Should be fixed now.

> One thing that thing that comes to mind re the areas that we send back an error over a channel: we should look into making a higher order function that takes a function and an error channel, then calls the function and returns the error over the channel.

Is this something we should do now or leave it for a follow-up?

Reply:
> Is this something we should do now or leave it for a follow-up?

I think a follow up, as we could use such a function in the wallet for the reservation flow as well.